### ISSUE TITLE
`update` event fires for all handles when one is changed with a key press

### ISSUE THREAD

**Issue by eoghanmurray** [leongersen/noUiSlider#960]
With multiple handles, moving one of them using the keyboard causes an `update` event to fire on all 3 handles.
```
noUiSlider.create(slider_el, {
    start: [30, 60, 90],
    behaviour: 'drag',
    connect: true,
    tooltips: [true, true, true],
    range: {
	'min': 0,
	'max': 100
    }
});
slider_el.noUiSlider.on('update', 
    function(v, h, u, t, p) { if (!t) { console.log('update:', h); }});
slider_el.noUiSlider.on('set', 
    function(v, h, u, t, p) { if (!t) { console.log('set:', h); }}
```

Expected output upon moving handle number 2 with keyboard:

```
update: 1
set: 1
```

Actual output:

```
update: 0
update: 1
set: 1
update: 2
```

This causes a bug with my solution in #957 when trying to move handle number 2 back past the margin of handle number 1; the event handler for handle number 1 gets executed first and modifies the position of handle number 2 such that when the `update` event for handle number 2 subsequently fires, there's nothing to do.

**Comment by eoghanmurray** [comment]
Looks like the rationale for the multiple updates was already questioned here:
https://github.com/leongersen/noUiSlider/issues/579#issuecomment-340594889
I know there's gonna be a good reason but I can't figure it out.
Could a solution be to check whether a given handles' value has actually changed before firing that event?

**Comment by leongersen** [comment]
This is a mistake on my end, `update` should only fire for the handle being, you know, updated.

**Comment by leongersen** [comment]
As per the rationale behind firing update events:

The idea is that calling `set` allows handles to "push" each other forward. Say you have a slider with range `0 - 10`, step `3`, start `1 - 6`. Calling `set([9, null])` causes handle `0` to move to `9`, forcing handle `1` to move to `9` as well, even though no update of handle `1` was requested.

Thanks for pointing this out, because I'm now realizing that keyboard interaction allows for this same effect: handle `0` can push handle `1` forward, but it shouldn't!

**Comment by leongersen** [comment]
This is fixed in noUiSlider 13.1.3. Many thanks for contributing, @eoghanmurray! Much appreciated.

**Comment by eoghanmurray** [comment]
Chuffed to have turned out to be useful!
